src/utils/kbn_field_types.js

```diff
@@ -0,0 +1,105 @@
+
+export class KbnFieldType {
+  constructor({ name, sortable, filterable, esTypes }) {
+    this.name = name;
+    this.sortable = !!sortable;
+    this.filterable = !!filterable;
+    this.esTypes = Object.freeze(esTypes || []);
+    Object.freeze(this);
```

Comment:
Freezing the field type instances is basically the whole reason this class exists, and it's only exported so it can be used in the tests, not so it can be subclassed. If that was desired the extensions would have to comply (attach getters/setters/methods to the prototype) or update to the `KbnFieldType` class. I'm fine with that.